### PR TITLE
No schedule on public holidays

### PR DESCRIPTION
**Pull request by patrixr**
### What does this PR do?

BUG: We suspect some timezone issue allows people to schedule deliveries on public holidays, this adds a backend check to ensure this never happens.

NOTE: I originally went with a model validation, but then realized the controller makes a ggv modification straight away, so I sneaked the verification in the controller to avoid that happening.

### Impacted Areas

Donor app > Transport schedule

### CODE REVIEW

File: app/controllers/api/v1/deliveries_controller.rb [hunk 1]
```diff
@@ -147,6 +149,30 @@ def get_delivery_details
...
+        begin
+          Date.parse(scheduled_at)
+        rescue ArgumentError
+          nil
+        end
+      end
+
+      def validate_schedule
+        if scheduled_date.nil?
+          render_error(I18n.t('schedule.bad_date'));
```
Comment (namrataukirde) [inline app/controllers/api/v1/deliveries_controller.rb]: Semicolon is not required.

File: app/controllers/api/v1/deliveries_controller.rb [hunk 2]
```diff
@@ -147,6 +149,30 @@ def get_delivery_details
...
+          return false
+        end
+
+        if Holiday.is_holiday(scheduled_date)
```
Comment (namrataukirde) [inline app/controllers/api/v1/deliveries_controller.rb]: can we add a question mark at the end of method name `is_holiday?`

**Comment by namrataukirde** [comment]
@patrixr, I feel we should not allow admin's as well to do booking on holiday's but we can ask Matt's thoughts on this.

File: app/controllers/api/v1/deliveries_controller.rb [hunk 3]
```diff
@@ -147,6 +149,30 @@ def get_delivery_details
...
+        return nil unless scheduled_at.present?
+        begin
+          Date.parse(scheduled_at)
+        rescue ArgumentError
+          nil
+        end
+      end
+
+      def validate_schedule
+        if scheduled_date.nil?
```
Comment (namrataukirde) [inline app/controllers/api/v1/deliveries_controller.rb]: Is there any possibility that schedule_date can contain a blank string? 🤔

Comment (swatijadhav) [reply app/controllers/api/v1/deliveries_controller.rb]: `.blank?` will be safer to use.

Comment (patrixr) [reply app/controllers/api/v1/deliveries_controller.rb]: `scheduled_date` is a method that returns `nil?` explicitely

**Comment by patrixr** [comment]
> @patrixr, I feel we should not allow admin's as well to do booking on holiday's but we can ask Matt's thoughts on this.

Cool, I'll move forward with this and discuss later

**Pull request merged**